core/bind/core_bind.cpp

```diff
@@ -483,7 +483,7 @@ String _OS::get_name() const {
 }
 Vector<String> _OS::get_cmdline_args() {
 
-	List<String> cmdline = OS::get_singleton()->get_cmdline_args();
+	List<String> cmdline = OS::get_singleton()->get_project_cmdline_args();
```

Comment:
I'm not sure if it should be like this. This way calling OS.get_cmdline_args() from gds would return the arguments for the game.
For example in: `godot --quiet -- test 55 3`
get_cmdline_args() would return` ["test", "55", "3"]`

But the docs say get_cmdline_args returns "the command line arguments passed to the engine".
Maybe it should just pass all the commands. The alternative is exposing `get_project_cmdline_args` (which I implemented in the class OS) so we have both, all the arguments game+engine (get_cmdline_args) and game arguments only (get_project_cmdline_args).

Reply:
This could break compat imo, there could be some editor tools which rely on engine-specific args, I'd just expose `get_project_cmdline_args` separately from `get_cmdline_args`. Yet I'm still not sure how these project-specific args get registered by the parser currently.

> Maybe it should just pass all the commands.

This could be a decent trade-off, + exposing `CommandParser` to scripting would help developers to sort out engine-specific from project-specific args. What about "black-listing" (or rather reserving) engine-specific args?

Reply:
As a side note, here's how I emulate project-specific args from within project settings, editor-only:

![godot_main_loop_args](https://user-images.githubusercontent.com/17108460/65877410-4b546d80-e394-11e9-912b-1ffa675d6ae7.png)